src/slice_cursor.rs

```diff
@@ -0,0 +1,164 @@
+use std::{
+    fmt::Debug,
+    io::{Cursor, Seek as _},
+};
+
+use crate::error::Error;
+
+#[derive(Default, Eq, PartialEq)]
+pub(crate) struct BinarySliceCursor<'a> {
+    inner: &'a [u8],
+    pos: usize,
+}
+
+impl<'a> BinarySliceCursor<'a> {
+    /// Creates a new cursor from a binary slice
+    pub(crate) fn new(slice: &'a [u8]) -> Self {
+        Self {
+            inner: slice,
+            pos: 0,
+        }
+    }
+
+    /// Gets current cursor position
+    pub fn pos(&self) -> usize {
```

Comment:
```suggestion
    pub fn current_cursor_position(&self) -> usize {
```

The function length doesn't matter, and it will be to easier to work with it than checking the doc comment

Reply:
`current_cursor_position`

There is no other position or cursor then the current one, so `current` is pointless, and `cursor` is just repeating the type name. 
This is like renaming `len()` in std `Vec` to `current_vector_length` 🤦. 
I can extend it to `position` if you want, but I don't think adding those words to the name make it any more clear, then it already is.

Reply:
Yeah ik that, but I just used it to show you that there is no need of writing `Gets current cursor position`, because as you said:

> so current is pointless, and cursor is just repeating the type name.

Reply:
A sentence is easier and faster to read and understand, then a line of code / function definition.

btw. we bikeshedded on this already some time ago, and no one managed to convince the other side 😅

Reply:
fjaskdljaskldj wait or 0.0.7 and you'll see

Reply:
wat?